tools/export/cdt/__init__.py

```diff
@@ -25,15 +56,31 @@ def generate(self):
             'include_paths': [starting_dot.sub('%s/' % self.project_name, inc) for inc in self.resources.inc_dirs],
             'load_exe': str(self.LOAD_EXE).lower()
         }
-
+        
+        launch_cfgs = {}
+        for launch_name in supported_launches:
+            launch = dict(ctx.items())
+            launch.update({'device': self.get_target_config(ctx, launch_name)}.items())
+            launch_cfgs.update({launch_name: launch})
+            
         if not exists(join(self.export_dir,'eclipse-extras')):
             makedirs(join(self.export_dir,'eclipse-extras'))
 
+        for launch_name, ctx in launch_cfgs.items():
+            # Generate launch configurations for former GNU ARM Eclipse plug-in
+            self.gen_file('cdt/%s' % 'pyocd_settings_gnu_arm.tmpl', ctx, join('eclipse-extras',
+                            '{target}_{project}_{conf}_{launch}.launch'.format(
```

Comment:
```suggestion
                            '{target}_{project}_{conf}_pyocd_settings.launch'.format(
```

And drop the `launch='pyocd_settings'` below